app/src/main/java/com/ekeitho/reduxy/ViewModel.kt

```diff
@@ -1,27 +1,43 @@
 package com.ekeitho.reduxy
 
 import android.databinding.ObservableField
+import android.view.View
 import io.reactivex.Observable
 import io.reactivex.disposables.Disposable
 import io.reactivex.disposables.Disposables
-import io.reactivex.subjects.BehaviorSubject
-
+import io.reactivex.subjects.Subject
 
 object ViewModel {
 
     val nameObs = ObservableField<String>()
     val ageObs = ObservableField<Int>()
     val isHappy = ObservableField<Boolean>()
+    lateinit var eventStream : Subject<Event>
 
     var subscription : Disposable = Disposables.disposed()
 
-    fun init(state: Observable<ApplicationState>) {
+    fun onNameChanged(s : CharSequence, start : Int, before : Int, count : Int) {
+        eventStream.onNext(NameChangeEvent(s.toString()))
+    }
+
+    fun onAgeChanged(s : CharSequence, start : Int, before : Int, count : Int) {
+        // If parsing this CharSequence to Integer fails, it means that
+        // the edittext has not been set to number input, and therefore
+        // we should crash
+        eventStream.onNext(AgeChangeEvent(Integer.parseInt(s.toString())))
+    }
+
+    fun onButtonClicked(v : View?) {
+        eventStream.onNext(ButtonClickEvent())
+    }
```

Comment:
onNameChanged and onAgeChanged needed to be called on the VM, so I moved onButtonClicked for consistency.